### DISCUSSION TITLE
Post-installation method, no login to Cockpit possible (Wrong user name or password)

### DISCUSSION THREAD

**Discussion by HachimanSec** [telekom-security/tpotce#1080] [Q&A]
Hello, 

I hope I didn't miss any thread, but I searched the forums and didn't find anything. 

I am trying to setup a test deployment of T-pot. The installation succeeds, however I am unable to login to the cockpit. Installation was done via [post install method](https://github.com/telekom-security/tpotce#post-install).

**There is always the error of _Wrong user name or password_ for the Cockpit website login.**

However, the OS username and password combi is correct and still works via SSH login. 
Login to `<IP>: 64297` works, via username and password I created during installation. 

I explicitly read [First start](https://github.com/telekom-security/tpotce/blob/master/README.md#first-start) and [user types](https://github.com/telekom-security/tpotce/blob/master/README.md#user-types).

Anyone any ideas? 

Thanks!

Below are my debug informations:
> T-pot version: T-Pot 22.04.0, latest via git as of 2022-04-26
> T-pot edition: Standard
> Architecture: Cloud
> Hoster: DigitalOcean
> OS: Debian 11 x64
> CPU: 4 vCPUs
> RAM: 8GB / 160GB Disk
> Install errors: None
> df -h:
```
Filesystem      Size  Used Avail Use% Mounted on
udev            3.9G     0  3.9G   0% /dev
tmpfs           796M  3.8M  792M   1% /run
/dev/vda1       160G   12G  142G   8% /
```
> lsb_release -a && uname -a
```
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 11 (bullseye)

Release:	11
Codename:	bullseye
Linux certainaugust 5.10.0-13-amd64 #1 SMP Debian 5.10.106-1 (2022-03-17) x86_64 GNU/Linux
```
> systemctl status tpot:
```
Loaded: loaded (/etc/systemd/system/tpot.service; enabled; vendor preset: enabled)
     Active: active (running) since Tue 2022-04-26 15:28:40 UTC; 4min 11s ago
```

<img width="1788" alt="Screenshot 2022-04-26 at 17 32 45" src="https://user-images.githubusercontent.com/20123996/165337831-610acdbc-9c7e-44fe-84b3-962f8f304eeb.png">
<img width="1788" alt="Screenshot 2022-04-26 at 17 30 41" src="https://user-images.githubusercontent.com/20123996/165337849-b0db9639-72af-4da7-8952-f7d5a13072cb.png">
<img width="1788" alt="Screenshot 2022-04-26 at 17 30 29" src="https://user-images.githubusercontent.com/20123996/165337854-948e9070-ebb5-4bbb-b978-7a0d24760240.png">

**Comment by HachimanSec** [comment]
I actually found another thread: https://github.com/telekom-security/tpotce/discussions/945

The cockpit apparently requires _tsec_ as `<os_username>` (to stick with the nomenclature of [first start](https://github.com/telekom-security/tpotce/blob/master/README.md#first-start). Other OS usernames are not accepted.) 

SSH'ing into the server and adding a new user named _tsec_ solves the problem. 
`adduser tsec`

Either the documentation, the installer or the Cockpit access config could be updated to address this behaviour. 

EDIT: The newly added tsec user can login to the Cockpit, however it is not admin. Root is admin, but I can't login. Anyone any idea how I can manually promote tsec to Cockpit admin? I cant find anything in the doc.

**Reply by HachimanSec** [reply]
PS: Interestingly the original OS account _root_ is listed in the Cockpit _Accounts_ menu. It simply doesnt work for the login. 

As mentioned above, the problem is that the newly added user is not an admin. Root is admin, but can't login and it seems one can't make an os user a Cockpit admin via CLI. 

<img width="1126" alt="Screenshot 2022-04-26 at 17 45 39" src="https://user-images.githubusercontent.com/20123996/165340189-7c13a59a-9d74-4853-b6cf-968686452ee0.png">

**Reply by t3chn0m4g3** [reply]
Yes, usually a Debian Install offers to add a standard user during install and this user is then added to Cockpit. However this seems to be different in some cases. Since Cockpit is a system service it adds `root` for elevation.

**Comment by t3chn0m4g3** [comment]
Will look into it. Probably related to #1074.

**Reply by HachimanSec** [reply]
Danke dir @t3chn0m4g3. I just update my comments with some more info.

**Reply by t3chn0m4g3** [reply]
Welcome @HachimanSec and thanks for reporting!

**Reply by frogman1370** [reply]
I know this thread is a couple years old and I was looking into the same thing, but in reviewing the docs for 22.04:

<html>
<body>

64294 | tcp | incoming | T-Pot Management: Sensor data transmission to hive (through NGINX reverse proxy) to 127.0.0.1:64305
-- | -- | -- | --



</body>
</html>

I'm assuming that this Cockpit is only meant for a distributed model where it uses the web interface to send data?  Let me know if I'm incorrect on that assumption.  I created a 'tsec' user (rebooted) and still unable to access the interface.
